src/web3/KeychainWallet.ts

```diff
@@ -1,16 +1,22 @@
+import { Address } from '@celo/connect'
 import { normalizeAddressWith0x, privateKeyToAddress } from '@celo/utils/lib/address'
 import { UnlockableWallet } from '@celo/wallet-base'
 import { RemoteWallet } from '@celo/wallet-remote'
 import { ErrorMessages } from 'src/app/ErrorMessages'
 import Logger from 'src/utils/Logger'
-import { ImportMnemonicAccount, KeychainSigner, listStoredAccounts } from 'src/web3/KeychainSigner'
+import ContractKitSigner from 'src/web3/ContractKitSigner'
+import { KeychainAccountManager, listStoredAccounts } from 'src/web3/KeychainAccountManager'
+import { ImportMnemonicAccount } from 'src/web3/types'
 
 const TAG = 'web3/KeychainWallet'
 
 /**
  * A wallet which uses the OS keychain to store private keys
+ * It implements many methods necessary to be an UnlockableWallet so it can be used in ContractKit
+ * Meant to be used as a singleton and helps manage multiple accounts (addresses) stored on the Keychain
  */
-export class KeychainWallet extends RemoteWallet<KeychainSigner> implements UnlockableWallet {
+export class KeychainWallet extends RemoteWallet<ContractKitSigner> implements UnlockableWallet {
```

Comment:
Yes KeychainWallet is very contractkit specific. This is by necessity since its [loaded directly into contractkit](https://github.com/valora-inc/wallet/blob/main/src/web3/contracts.ts#LL80C12-L80C12) and has to comply with the spec of UnlockableWallet